### PR TITLE
chore: add theme provider

### PR DESCRIPTION
**Pull request by shanberg**

### CODE REVIEW

**Comment by tilleryd** [comment]
I've done the following to get this PR up to date with master, tests running, and storybook running:

- merged master and fixed conflicts
- commented out code using `transparentize` and `darken` from `polished`, not sure why they aren't working, could be a dependency version issue
- commented out tests in `Spinner.test.js` and disabled test in `generateButtonStyles.test.js`
- kept the usage of `them.brandColor` because I didn't want to refactor in this branch

File: src/components/Button/components/ButtonWrap/ButtonWrap.js [hunk 1]
```diff
@@ -55,13 +44,15 @@ const ButtonWrap = styled.button`
...
     border: 1px solid
-      ${({ theme }) => theme.brandColor || COLOR_INTENT_HIGHLIGHT};
-    box-shadow: ${({ theme }) => css`0 0 0 ${FORM_HIGHLIGHT_SIZE}px
-        ${transparentize(
-          1 - OPACITY_50,
-          theme.brandColor || COLOR_INTENT_HIGHLIGHT
-        )},
-      inset 0 0 0 1px ${transparentize(1 - OPACITY_15, COLOR_BACKGROUND_A)}`}};
+      ${({ theme }) => theme.brandColor || theme.COLOR_INTENT_HIGHLIGHT};
+    /* box-shadow: ${({ theme }) => css`0 0 0 ${theme.FORM_HIGHLIGHT_SIZE}px
```
Comment (kaitmore) [inline src/components/Button/components/ButtonWrap/ButtonWrap.js]: Should this block be commented out?

File: src/components/Button/components/ButtonWrap/utils/generateButtonStyle.js [hunk 2]
```diff
@@ -37,35 +29,40 @@ function generateButtonTypeColors(type) {
...
+function generateButtonStyle(theme, type, renderBorder) {
+  const baseColor = generateButtonTypeColors(theme, type);
 
-  return `
-  background-color: ${baseColor};
-  color: ${contrastColor(baseColor, 1)};
-  border-color: ${renderBorder ? darken(0.1, baseColor) : `transparent`};
+  const colorStyles = css`
+    background-color: ${baseColor};
+    /* border-color: ${renderBorder ? darken(0.1, baseColor) : `transparent`};
```
Comment (kaitmore) [inline src/components/Button/components/ButtonWrap/utils/generateButtonStyle.js]: more commented out code ^^

File: src/components/Button/components/ButtonWrap/ButtonWrap.js (hunk 1 above)
Comment (tilleryd) [reply src/components/Button/components/ButtonWrap/ButtonWrap.js]: No, I noted this in my comment. The `transparentize` function was throwing an error.

Comment (kaitmore) [reply src/components/Button/components/ButtonWrap/ButtonWrap.js]: Sorry! Missed your comments

**Comment by kaitmore** [comment]
- Decipher logo in the AppFooter (Keen Dark)
image

- Icon names don't show up when switching to KeenDark
image

- NavCard looks a little off in KeenDark
image

- Should NavCardGroup have a black background in KeenDark?
image

- Spinner is broken in both themes:
image

**Comment by shanberg** [comment]
- Fixed missing styles in Spinner
- Fixed glyph labels

**Comment by shanberg** [comment]
@kaitmore I think the NavCard can stay the way it is for now. I think there's room for discussion about how the background and content colors should be treated, but it shouldn't block merging the rest of this branch.

**Comment by kaitmore** [comment]
@tilleryd @shanberg I think we should have fallback variables for all of the `props.theme.<THEME_VAR>` declarations in the code. Otherwise if you remove the theme, everything breaks. We could use Keen as the default, or create some non-branded base theme. Either way, I don't think the lib should totally break if the developer doesn't use the `ThemeProvider`.

**Comment by shanberg** [comment]
Default theme should probably be `keen`. We shouldn't ever need a basic theme because we're always either creating a theme to serve some custom use of the library or we're building a straight Grey Matter app.

**Comment by kaitmore** [comment]
- added defaultProps that set a fallback theme to all styled-components that use theme variables
- deleted top level Checkbox component which wasn't being used anywhere
- fixed a few failing tests
- removed any manual passing of the theme prop, in tests and in the component code
- gave ProptypesTable a default value because it was throwing an error

To test:
- comment out the theme provider in `.storybook/config.js`
- start storybook and confirm that all components use the `keen` theme

**Pull request merged**